dotcom-rendering/src/web/components/SnapWrapper.tsx

```diff
@@ -0,0 +1,27 @@
+import { css } from '@emotion/react';
+import type { DCRSnapType } from '../../types/front';
+
+type Props = {
+	snapData?: DCRSnapType;
+	children?: React.ReactNode;
+};
+
+export const SnapWrapper = ({ snapData, children }: Props) => {
+	if (snapData?.embedHtml === undefined) {
+		return <></>;
+	}
+
+	return (
+		<>
+			<div
+				css={[
+					css`
+						${snapData.embedCss};
+					`,
+				]}
```

Comment:
Definitely worth adding a comment here explaining the 'sandboxing', how it works, why we're doing it, etc !

Reply:
Makes sense! I will try to give my understanding of how emotion allows us to do the 'sandboxing' below:

In DCR we are using [emotion](https://emotion.sh/) to apply styles to our components. You can read more about the decision to use it [here](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/docs/architecture/004-emotion.md). One of the reasons it was chosen is because it allows us to scope styles to a component. In the case of thrashers here is how it works and why it is important for us:

![image](https://github.com/guardian/dotcom-rendering/assets/19683595/0ff82564-1de1-4005-8c4e-ecf3da147f31)

* The `css` function gets called with some styles, in this case `${snapData.embedCss}`
* The `css` function generates a random class name, e.g. `.dcr-tyelbh-SnapCssSandbox`, 
* The computed class name and the styles are passed to the `css` prop 
* The styles are evaluated and the computed class name is eventually applied to the `className` prop under the hood.
* The styles get applied to the component only with the particular `className`

This mechanism allows us to 'sandbox' the CSS and ..

WIP